src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -337,90 +329,114 @@ def crop_augmenter(image,
-                    # Normalize the elements to the cropped width and height  
+
+                    # Normalize the elements to the cropped width and height
                     ele_1 = t_intersection[1] / cropped_width
                     ele_2 = t_intersection[0] / cropped_height
                     ele_3 = (t_intersection[3] - t_intersection[1]) /cropped_width
                     ele_4 = (t_intersection[2] - t_intersection[0]) / cropped_height
-                    
+
                     formatted_annotation.append(np.concatenate([identifier, np.array([ele_1, ele_2, ele_3, ele_4]), confidence]))
                 else:
                     formatted_annotation.append(np.concatenate([identifier, np.array([0.0, 0.0, 0.0, 0.0]), confidence]))
             else:
                 formatted_annotation.append(np.concatenate([identifier, np.array([0.0, 0.0, 0.0, 0.0]), confidence]))
-        
-        
+
+
         if not is_min_object_covered:
             continue
 
         y_offset = int(y_offset)
         x_offset = int(x_offset)
         end_y = int(cropped_height + y_offset)
         end_x = int(cropped_width + x_offset)
-            
+
         image_cropped = image[y_offset:end_y, x_offset:end_x]
-            
+
         return np.array(image_cropped), np.array(formatted_annotation, dtype=np.float32)
-    
+
     return np.array(image), annotation
 
-def complete_augmenter(img_tf, ann_tf, output_height, output_width):
-    img_tf, ann_tf = tf.numpy_function(func=crop_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = tf.numpy_function(func=padding_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = tf.numpy_function(func=horizontal_flip_augmenter, inp=[img_tf, ann_tf], Tout=[tf.float32, tf.float32])
-    img_tf, ann_tf = color_augmenter(img_tf, ann_tf)
-    img_tf, ann_tf = hue_augmenter(img_tf, ann_tf)
+def complete_augmenter(img_tf, ann_tf, output_height, output_width, seed_tf):
+    img_tf, ann_tf = crop_augmenter_wrapper(img_tf, ann_tf, seed_tf)
+    img_tf, ann_tf = padding_augmenter_wrapper(img_tf, ann_tf, seed_tf)
+    img_tf, ann_tf = horizontal_flip_augmenter_wrapper(img_tf, ann_tf, seed_tf)
+    img_tf, ann_tf = color_augmenter(img_tf, ann_tf, seed_tf)
+    img_tf, ann_tf = hue_augmenter(img_tf, ann_tf, seed_tf)
     img_tf, ann_tf = resize_augmenter(img_tf, ann_tf, (output_height, output_width))
+
+    return img_tf, ann_tf
+
+def crop_augmenter_wrapper(img_tf, ann_tf, seed_tf):
+
+    max_attempts = _DEFAULT_AUG_PARAMS["max_attempts"]
+    random_nums = tf.random.uniform([1+4*max_attempts], 0., 1., seed=seed_tf)
+    img_tf, ann_tf = tf.numpy_function(func=crop_augmenter, inp=[img_tf, ann_tf, random_nums], Tout=[tf.float32, tf.float32])
+    return img_tf, ann_tf
+
+def padding_augmenter_wrapper(img_tf, ann_tf, seed_tf):
+
+    max_attempts = _DEFAULT_AUG_PARAMS["max_attempts"]
+    random_nums = tf.random.uniform([1+max_attempts+3], 0., 1., seed=seed_tf)
```

Comment:
Is there correlation between the `random_nums` values computed in this function versus in e.g. `crop_augmenter_wrapper`? Note that the first value of `random_nums` is often used to evaluate the skip probability. So if these are correlated, then whether or not each augmentation is performed will be improperly correlated.

Reply:
Good!